### DISCUSSION TITLE
关于文档相关的内容...

### DISCUSSION THREAD

**Discussion by mengban1110** [simple-robot/Discussions#76] [Q&A] [answered]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

第一次接触qq机器人, 了解了一下咱们的文档, 比如我想实现一个, 群内@我 的 相关功能, 请问有类似事件吗? 并且看咱们的文档有些不知道怎么实现某些东西, 感觉有点迷糊...是功力不到么,有什么好的建议吗... 希望有些小功能demo?

**Comment by mengban1110** [comment]
并且有一些功能上使用的小疑问, 比如MessageBuilder里append消息 如何实现换行等等 :(

**Comment by ForteScarlet** [comment, marked as answer]
精力和文笔有限，文档确实不能写的面面俱到而且的确不尽人意。

跟at有关的内容，如果是指at BOT后触发事件，参考 [文档@Filter.Targets](https://simbot.forte.love/docs/basic/event-listening#%E7%9B%AE%E6%A0%87%E8%BF%87%E6%BB%A4) 中的 `atBot` 属性

如果是指at某个其他人，需要在事件的 `messageContent.messages` 中自行寻找获取 [`At`](https://simbot.forte.love/docs/definition/message-overview/standard-message#at) 类型的消息对象，然后获取并判断属性（比如at的目标是否为需要的）然后编写逻辑

> MessageBuilder里append消息 如何实现换行等等

换行就是正常的文本换行符 `\n` ，比如 `Text.of("第一行\n第二行")`

**Reply by mengban1110** [reply]
非常感谢您半夜回复我, 这个属性我在文档中看见了, 简单尝试了一下发现没有什么效果, 应该是用的不太对, 我再试试。还有就是咱们的QQ群说在git首页, 貌似没看见呀...我非常统一建设社区的想法, 但是对于初学者来讲, 快速上手沟通可能更有理由发展吧 ❤

**Reply by ForteScarlet** [reply]
群号在 [#交流与反馈](https://github.com/simple-robot#-%E4%BA%A4%E6%B5%81%E4%B8%8E%E5%8F%8D%E9%A6%88) 节下

**Reply by ForteScarlet** [reply]
atBot最好配合 `@ContentTrim` 使用。

**Reply by mengban1110** [reply]
非常感谢@ContentTrim的tips, 这个注解还是B站有人发demo看见的, 没用这个 atBot就没生效, 现在好了, 十分感谢半夜帮助🙏🏻
